ivc/src/poseidon_params_8_56_5_3.rs

```diff
@@ -979,9 +979,9 @@ impl SpongeConstants for PlonkSpongeConstantsIVC {
     const SPONGE_CAPACITY: usize = 1;
     const SPONGE_WIDTH: usize = 3;
     const SPONGE_RATE: usize = 2;
-    const PERM_ROUNDS_FULL: usize = 8;
+    const PERM_ROUNDS_FULL: usize = 0;
```

Comment:
I want to understand this better: this poseidon is `8_56`, so it's 4 full rounds, 56 partial, and 4 again. Why is `PERM_ROUNDS_FULL=0` then? And what are "HALF_ROUNDS_FULL?"...

Reply:
`SpongeConstants` is undocumented :(

Reply:
Ah OK I see now:
* If `SC::PERM_HALF_ROUNDS_FULL > 0` we're in half-round mode, then Poseidon will do `PERM_HALF_ROUNDS_FULL` full rounds, then `PERM_ROUNDS_PARTIAL` partial rounds, then again `PERM_HALF_ROUNDS_FULL` ones
* Otherwise we're in full-round mode and Poseidon will do `PERM_ROUNDS_FULL` full rounds.